src/server/scriptInfo.ts

```diff
@@ -119,6 +134,23 @@ namespace ts.server {
             this.pendingReloadFromDisk = true;
         }
 
+        public getFileSize(): number {
+            return this.useScriptVersionCacheIfValidOrOpen()
+                ? this.svc!.getSnapshot().getLength()
+                : (this.fileSize || this.getOrLoadText().length);
+        }
+
+        public getTelemetryFileSize(): number {
+            // This is only for telemetry, so a slightly stale value is fine.
+            return !!this.fileSize

```

Comment:
I think this is better and efficient version.
```ts
return this.text !== undefined ? thist.fileSize || this.text.length : Debug.assertDefined(this.svc).getSnapshot().getLength();
```

Reply:
It looks like that `assertDefined` fails if the size is requested immediately after construction of the `TextStorage`.  I'll check `text` before `svc` though.